staker/block_validator.go

```diff
@@ -796,6 +841,13 @@ func (v *BlockValidator) iterativeValidationPrint(ctx context.Context) time.Dura
 	}
 	log.Info("validated execution", "messageCount", printedCount, "globalstate", validated.GlobalState, "WasmRoots", validated.WasmRoots)
 	v.lastValidInfoPrinted = validated
+	if v.config().Dangerous.Revalidation.EndBlock > 0 && validated.GlobalState.Batch >= v.config().Dangerous.Revalidation.EndBlock {
+		if v.config().Dangerous.Revalidation.ThenQuit {
+			v.quitNode <- fmt.Sprintf("revalidation done from %d to %d", v.config().Dangerous.Revalidation.StartBlock, v.config().Dangerous.Revalidation.EndBlock)
+		} else {
+			v.StopAndWait()
```

Comment:
why we need this?

Reply:
There are basically two mode we wand to support if endblock is given to us.
If quit is enabled, stop the whole node after the validation reaches endblock
Else if quit is disabled, just stop the validation and not the whole node, so the StopAndWait is to stop BlockValidator.

Reply:
Done.